lib/tasks/reporting.rake

```diff
@@ -80,6 +80,47 @@ namespace :reporting do
     end
   end
 
+  desc "A CSV report of the number of publications in draft state by organisation between the dates specified [year-month-day-hour-min]"
+  task :number_of_draft_publications_by_organisation_by_date_range, %i[start_date end_date] => :environment do |_t, args|
+    if args[:start_date].present? && args[:end_date].present?
+      start_date = Time.parse(args[:start_date])
+      end_date = Time.parse(args[:end_date])
+    end
+    date_range = start_date...end_date
+
+    path = "#{Rails.root}/tmp/number_of_draft_publications_by_organisation_between_#{start_date}_to_#{end_date}.csv"
+
+    CSV_HEADERS = ["Lead publishing organisation", "Number of draft publications"].freeze
+
+    CSV.open(path, 'wb') do |csv|
```

Comment:
You can actually do `CSV.open(path, "wb", headers: CSV_HEADERS, write_headers: true)` here to avoid the line below.